streams/src/test/java/org/apache/kafka/streams/state/internals/ChangeLoggingKeyValueStoreTest.java

```diff
@@ -174,7 +173,7 @@ public void shouldQueryRange() throws Exception {
         store.put(hello, world);
         store.put(hi, there);
         store.put("zooom", "home");
-        final KeyValueIterator<String, String> range = store.range(hello, hi);
+        final KeyValueIterator<String, String> range = store.range(hello, "zooom");
         assertThat(range.next(), equalTo(KeyValue.pair(hello, world)));
         assertThat(range.next(), equalTo(KeyValue.pair(hi, there)));
         assertFalse(range.hasNext());
```

Comment:
I am confused. If we change `store.range(hello, "zooom");` should `hasNext()` not return `true` (start and end are inclusive). Thus, to me it seems the test is wrong and there must be a bug in the iterator code.